beacon-chain/sync/subscriber.go

```diff
@@ -475,110 +498,92 @@ func (s *Service) subscribeToSubnets(p subscribeToSubnetsParameters) error {
-				return
-			}
+	if err := s.subscribeToSubnets(subnetP); err != nil {
+		if errors.Is(err, errInvalidDigest) {
+			log.WithFields(logFields).Debug("Digest is invalid, stopping subscription")
+			return
 		}
-	}()
-
-	// Warn the user if we are not subscribed to enough peers in the subnets.
-	go func() {
-		log := log.WithField("minimum", minimumPeersPerSubnet)
-		logTicker := time.NewTicker(5 * time.Minute)
-		defer logTicker.Stop()
+		log.WithFields(logFields).WithError(err).Error("Could not subscribe to subnets")
+		return
+	}
 
-		for {
-			select {
-			case <-logTicker.C:
-				currentSlot := s.cfg.clock.CurrentSlot()
-				subnetsToFindPeersIndex := computeAllNeededSubnets(currentSlot, p.getSubnetsToJoin, p.getSubnetsRequiringPeers)
-
-				isSubnetWithMissingPeers := false
-				// Find new peers for wanted subnets if needed.
-				for index := range subnetsToFindPeersIndex {
-					topic := fmt.Sprintf(p.topicFormat, p.digest, index)
-
-					// Check if we have enough peers in the subnet. Skip if we do.
-					if count := s.connectedPeersCount(topic); count < minimumPeersPerSubnet {
-						isSubnetWithMissingPeers = true
-						log.WithFields(logrus.Fields{
-							"topic":  topic,
-							"actual": count,
-						}).Warning("Not enough connected peers")
-					}
-				}
+	ctx, cancel := context.WithTimeout(s.ctx, timeout)
+	defer cancel()
+	if err := s.cfg.p2p.FindAndDialPeersWithSubnets(ctx, p.topicFormat, p.digest, minPeers, neededSubnets); err != nil && !errors.Is(err, context.DeadlineExceeded) {
+		log.WithFields(logFields).WithError(err).Debug("Could not find peers with subnets")
+	}
+}
 
-				if !isSubnetWithMissingPeers {
-					log.WithField("topic", shortTopic).Info("All subnets have enough connected peers")
+func (s *Service) logMinimumPeersPerSubnet(p subscribeParameters, logFields logrus.Fields) {
+	minimumPeersPerSubnet := flags.Get().MinimumPeersPerSubnet
+	// Warn the user if we are not subscribed to enough peers in the subnets.
+	log := log.WithField("minimum", minimumPeersPerSubnet)
+	logTicker := time.NewTicker(5 * time.Minute)
+	defer logTicker.Stop()
+
+	for {
+		select {
+		case <-logTicker.C:
+			currentSlot := s.cfg.clock.CurrentSlot()
+			subnetsToFindPeersIndex := computeAllNeededSubnets(currentSlot, p.getSubnetsToJoin, p.getSubnetsRequiringPeers)
+
+			isSubnetWithMissingPeers := false
+			// Find new peers for wanted subnets if needed.
+			for index := range subnetsToFindPeersIndex {
+				topic := fmt.Sprintf(p.topicFormat, p.digest, index)
+
+				// Check if we have enough peers in the subnet. Skip if we do.
+				if count := s.connectedPeersCount(topic); count < minimumPeersPerSubnet {
+					isSubnetWithMissingPeers = true
+					log.WithFields(logrus.Fields{
+						"topic":  topic,
+						"actual": count,
+					}).Warning("Not enough connected peers")
 				}
-
-			case <-s.ctx.Done():
-				return
 			}
+			if !isSubnetWithMissingPeers {
+				log.WithFields(logFields).Info("All subnets have enough connected peers")
```

Comment:
I know it's not in this PR, but is it possible to move from Info to Debug?
(I introduced this log myself, and I regretted it to have it in Info. The rationale is there is no need to pollute info log console if everything is allright.)